### PR TITLE
Add Python 3.7 and 3.8 support

### PR DESCRIPTION
**Pull request by ecederstrand**
2.6 is supported according to code in setup.py, so add that.

Refs #1508

### CODE REVIEW

**Comment by ecederstrand** [comment]
Hmm. Tests are failing, even on Python 3.6 that I didn't touch in this MR. Seems unrelated changes are necessary to get the test suite to succeed again.

**Comment by hainm** [comment]
thanks @ecederstrand. The tests failed due to mpi testing. It seems to me that travis mpirun does not support bellow syntax anymore

```
+++mpirun -n 2 python mpi_molsurf.py
--------------------------------------------------------------------------
There are not enough slots available in the system to satisfy the 2
slots that were requested by the application:

  python

Either request fewer slots for your application, or make more slots
available for use.

A "slot" is the Open MPI term for an allocatable unit where we can
launch a process.  The number of slots available are defined by the
environment in which Open MPI processes are run:

  1. Hostfile, via "slots=N" clauses (N defaults to number of
     processor cores if not provided)
  2. The --host command line parameter, via a ":N" suffix on the
     hostname (N defaults to 1 if not provided)
  3. Resource manager (e.g., SLURM, PBS/Torque, LSF, etc.)
  4. If none of a hostfile, the --host command line parameter, or an
     RM is present, Open MPI defaults to the number of processor cores

In all the above cases, if you want Open MPI to default to the number
of hardware threads instead of the number of processor cores, use the
--use-hwthread-cpus option.

Alternatively, you can use the --oversubscribe option to ignore the
number of available slots when deciding the number of processes to
launch.
--------------------------------------------------------------------------
+++set +x
```

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,8 @@ matrix:
     - { os: linux, env: PYTHON_VERSION=2.7 }
     - { os: linux, env: PYTHON_VERSION=3.6 }
```
Comment (hainm) [inline .travis.yml]: may be drop python 3.6 testing

File: .travis.yml [hunk 2]
```diff
@@ -6,7 +6,8 @@ matrix:
...
     - { os: linux, env: PYTHON_VERSION=3.6 USE_OPENMP=false }
```
Comment (hainm) [inline .travis.yml]: may be drop python 3.6 testing and use `USE_OPENMP=false` for python 3.8

File: setup.py [hunk 3]
```diff
@@ -236,10 +236,13 @@ def prepare_env_for_osx():
             'Operating System :: MacOS',
             'Intended Audience :: Science/Research',
             'License :: OSI Approved :: GNU General Public License v3 (GPLv3)',
+            'Programming Language :: Python :: 2.6',
```
Comment (hainm) [inline setup.py]: I have intention to drop 2.6, 2.7, 3.4, 3.5 and only keep 2.7 and >= 3.7 so people can move one.

File: .travis.yml (hunk 1 above)
Comment (ecederstrand) [reply .travis.yml]: Done

File: .travis.yml (hunk 2 above)
Comment (ecederstrand) [reply .travis.yml]: Done

File: setup.py (hunk 3 above)
Comment (ecederstrand) [reply setup.py]: In my own packages I try to support the currently supported Python versions. If you don't mind, I think it's better to remove 3.5 and 3.6 only when these versions are also EOL, or this package contains code that is not compatible with these versions (use of f-strings, for example). There's no reason to deny people to use this package on 3.5 if it still works there.

I have remove the the EOL versions now.

File: .travis.yml [hunk 4]
```diff
@@ -2,12 +2,9 @@ language: cpp
 
 matrix:
   include:
-    # - { os: linux, env: PYTHON_VERSION=2.7 PYPY=true }
-    - { os: linux, env: PYTHON_VERSION=2.7 }
```
Comment (hainm) [inline .travis.yml]: Please keep python 2.7.

File: tests/test_run_mpi.py [hunk 5]
```diff
@@ -17,4 +17,4 @@
                     reason='require mpi4py')
 @pytest.mark.parametrize('pyfile', testlist)
 def test_all_mpi_scripts(pyfile):
-    subprocess.check_call(['mpirun', '-n', '4', 'python', pyfile])
+    subprocess.check_call(['mpirun', '--use-hwthread-cpus', 'python', pyfile])
```
Comment (hainm) [inline tests/test_run_mpi.py]: nice, thanks.

File: setup.py (hunk 3 above)
Comment (hainm) [reply setup.py]: ok, it's fine with me. So just keep python 2.7 (phenix package is still using python 2.7) and >=3.5

**Comment by ecederstrand** [comment]
I'm out of ideas as to why the mpirun tests are failing. I can't get any output in Travis except exit code 1. I don't use `pytraj` directly, just maintaing a programming environment that does, so I don't really know how to proceed. Do you have any hints? I just wanted to have `pytraj` binary pip install work on Python 3.8.

I can try to reproduce the environment created by the travis script, but can't promise when I'll get to it.

**Comment by ecederstrand** [comment]
Oh, I see the master branch is failing with exactly the same errors: https://travis-ci.org/Amber-MD/pytraj/builds/654860504

I'll wait for you to fix those issues before continuing on this branch.

**Comment by hainm** [comment]
Thank you. I will fix it as soon as I can.

**Comment by hainm** [comment]
@ecederstrand  Please rebase master (#1524). I am sorry for the long delay since I am pretty busy with my daily work now.

**Comment by ecederstrand** [comment]
Should be based on latest commit on your master branch.

**Comment by ecederstrand** [comment]
Hmm, the Python 3.8 non-openmp build is timing out: https://travis-ci.org/github/Amber-MD/pytraj/jobs/674742508. I'll add an additional build task to see if it is due to Python 3.8 or OpenMP.

**Comment by ecederstrand** [comment]
Not related to OpenMP. In both builds, we are stuck trying to to run the `test_pmap_hbond_with_solvent_bridge` test.

**Comment by hainm** [comment]
> In both builds, we are stuck trying to to run the test_pmap_hbond_with_solvent_bridge test.

Can you please skip this test for py3.8 for now?

**Comment by ecederstrand** [comment]
Skipped 3 test cases when Python version==3.8 and now tests pass. I don't have sufficient knowledge of the domain to see any common patterns, or possible solutions.

**Comment by hainm** [comment]
thanks @ecederstrand.

**Pull request merged**